lib/Pusher.php

```diff
@@ -533,7 +533,7 @@ public function socket_auth($channel, $socket_id, $custom_data = false)
         $this->validate_channel($channel);
         $this->validate_socket_id($socket_id);
 
-        if ($custom_data === true) {
```

Comment:
Fair point. I just did a little test here appending bools to strings. https://3v4l.org/JNIVr. true bools append, false don't. My only only other reason for the string test is this is a public function, so any old gubbins can be thrown in that argument and it'll be appended. 

It's a 'false' default, that's expected to be a string to append, but there's no validation that it's either of those.